### PR TITLE
Added comment to explain erroring workflow

### LINKED ISSUE #507: Nightly syncs are failing
### Thank you for reporting an issue.

- 

### OpenTofu Version

```shell
https://github.com/opentofu/registry/actions/workflows/generate-and-deploy.yml

These are failing with 

2024/05/13 00:23:49 ERROR : Got fatal error on delete: --max-delete threshold reached
2024/05/13 00:23:49 ERROR : Got fatal error on delete: --max-delete threshold reached
```

we should set a high max delete threshold to work around this.
```


### OpenTofu Configuration Files

-

### PR DESCRIPTION
**Pull request by Yantrio**
Closes #507

### CODE REVIEW

File: .github/workflows/generate-and-deploy.yml [hunk 1]
```diff
@@ -38,6 +38,10 @@ jobs:
         env:
           GH_TOKEN: ${{ secrets.GH_TOKEN }}
 
+      ## Note for developers consuming this: This action is sometimes expected to error
+      ## When the max-delete threshold is reached at 0. This is an indication that the
+      ## bucket is not in sync with the local files. This is expected behavior and is a signal that we should
+      ## manually run the generate-and-deploy-with-delete.yml workflow to ensure the bucket is in sync.
```
Comment (cube2222) [inline .github/workflows/generate-and-deploy.yml:44]: I would add that the sync was still supposed to happen before the error was raised, so new files and overrides should have succeeded.

**Review by cube2222 — APPROVED** [review]
with nit

File: .github/workflows/generate-and-deploy.yml (hunk 1 above)
Comment (Yantrio) [reply .github/workflows/generate-and-deploy.yml:44]: 👍

Comment (Yantrio) [reply .github/workflows/generate-and-deploy.yml:44]: added in https://github.com/opentofu/registry/pull/508/commits/a349ca528b96375cc470d858c1ca3a33ebee4ebe

**Review by ghost — APPROVED** [review]

**Pull request merged**